### DISCUSSION TITLE
Would it be possible to clear a table's query string with "clear filters" button?

### DISCUSSION THREAD

**Discussion by RX-77IT** [filamentphp/filament#6867] [Ideas]
Just a little request: would it be possible to clear a table's searchable() query string at the click of the "clear filters" button?

Thank you in advance for all your hard work!

**Comment by danharrin** [comment]
Check out the method that gets run on the LW component when the button is clicked, you can probably override and extend it

**Reply by RX-77IT** [reply]
Thank you for your answer.

In the end I decided to leave the "clear filters" button in the filters' popup as is. Instead, I added a button on the empty list with this custom action to clear filters and search query.
```php
protected function getTableEmptyStateActions(): array
{
    return [
        \Filament\Tables\Actions\Action::make('remove_all_filters')
            ->label('Clear all filters')
            ->icon('heroicon-s-filter')
            ->button()
            ->action(function (\Livewire\Component $livewire) {
                $livewire->resetTableFiltersForm();
                $livewire->fill(['tableSearchQuery' => '']);
                $livewire->updatedTableSearchQuery();
            }),
    ];
}
```
